solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,530 @@
+        external
+        ownerOnly
+        validFee(newBurnIncentiveNetworkFee)
+    {
+        emit BurnIncentiveFeeUpdated(
+            _burnIncentiveFee,
+            newBurnIncentiveNetworkFee,
+            _maxBurnIncentiveFeeAmount,
+            newMaxBurnIncentiveFeeAmount
+        );
+
+        _burnIncentiveFee = newBurnIncentiveNetworkFee;
+        _maxBurnIncentiveFeeAmount = newMaxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev returns the total amount of the burned governance tokens
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurnedAmount() external view returns (uint256) {
+        return _totalBurnedAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function vortex(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve conversion strategy
+        Strategy memory strategy = vortexStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), strategy.amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < strategy.paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = strategy.amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = strategy.paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _govToken) {
+                // if the source token is the governance token, don't try to convert it either, but rather include it in
+                // the amount to burn
+                //
+                // Please note that networkTokenConversionAmounts[i] will remain 0 in this case
+                totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(amount);
+            } else if (token == _networkToken) {
+                // if the source token is the network token, don't try to convert it, but rather add its amount to the
+                // total amount to convert to the governance token
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(amount);
+
+                networkTokenConversionAmounts[i] = amount;
+            } else {
+                if (token != NATIVE_TOKEN_ADDRESS) {
+                    // if the source token is a regular token, approve the converter to withdraw the token amount
+                    ensureAllowance(token, network, amount);
```

Comment:
technically we could do without the amount here either and just use balanceOf(burner) here as well. the overhead here will be a bit more gas, the benefit here is that we'll also swap tokens that were sent to the burner by mistake. thoughts?

Reply:
See my comment above (i.e., let's either fix all of them or neither?)

Reply:
In this case, I'd rather leave it as is, since the `amount` is already a result of a different `balanceOf` and this will generate another external call for every token